### PR TITLE
Initial code scaffolding for nimble-table component

### PR DESCRIPTION
**Pull request by mollykreis**
# Pull Request

## 🤨 Rationale

The `nimble-table` will be implemented in a large number of PRs. This submission puts some mostly empty files in place as a starting point for development of the table.

## 👩‍💻 Implementation

Create basic placeholder implementation for the table:
- Creates a `nimble-table` component that renders the string _Nimble table here_
- Create a Storybook story that shows the `nimble-table` component with a warning that the component is experimental
- Create unit test that verifies the `nimble-table` element can be created
- Create matrix tests

## 🧪 Testing

- Looked at the component in Storybook
- Ran the new unit test

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/nimble-components/src/table/tests/table-matrix.stories.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const component = (
+): ViewTemplate => html`
+    <nimble-table></nimble-table>
+`;
+
+export const tableThemeMatrix: Story = createMatrixThemeStory(
+    createMatrix(component)
+);
+
+export const hiddenTable: Story = createStory(
```
Comment (jattasNI) [inline packages/nimble-components/src/table/tests/table-matrix.stories.ts:39]: Many of our components also include a "Text customized" story that ensures that setting the component's font to be italics has an effect on rendering. I'm on the fence about including it now for the table too. Thoughts?

My initial thought: It seems unlikely a client would want the entire table contents to be italics. Rather they'd want to style individual cells or headers. This implies we'd probably have to do feature dev to make this work correctly, so we can wait until that's prioritized to add this test.

File: change/@ni-nimble-components-c9f2f270-d0d3-478a-b43a-c0777af5fc44.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (jattasNI) [inline change/@ni-nimble-components-c9f2f270-d0d3-478a-b43a-c0777af5fc44.json:2]: Could you update the component status table too? I think we use ⚠️ for in-work components.

Comment (jattasNI) [inline change/@ni-nimble-components-c9f2f270-d0d3-478a-b43a-c0777af5fc44.json:2]: In #830 I'm proposing that all changes to experimental/in-work components can be marked as `patch` to reduce churn on clients. Unless someone has a reason to disagree, I'd propose we start doing that even in this initial PR.

```suggestion
  "type": "patch",
```

**Review by jattasNI — APPROVED** [review]

File: change/@ni-nimble-components-c9f2f270-d0d3-478a-b43a-c0777af5fc44.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (jattasNI) [inline change/@ni-nimble-components-c9f2f270-d0d3-478a-b43a-c0777af5fc44.json:1]: Friendly reminder to approve the Chromatic checks before submitting.

Comment (mollykreis) [reply change/@ni-nimble-components-c9f2f270-d0d3-478a-b43a-c0777af5fc44.json:1]: Sure, I can do that. I've gotten mixed feedback on whether I should be doing that or whether there should be a reviewer that acts as a second set of eyes to make sure we're happy with the detected Chromatic changes.

File: change/@ni-nimble-components-c9f2f270-d0d3-478a-b43a-c0777af5fc44.json (hunk 2 above)
Comment (mollykreis) [reply change/@ni-nimble-components-c9f2f270-d0d3-478a-b43a-c0777af5fc44.json]: Updated

File: change/@ni-nimble-components-c9f2f270-d0d3-478a-b43a-c0777af5fc44.json (hunk 3 above)
Comment (jattasNI) [reply change/@ni-nimble-components-c9f2f270-d0d3-478a-b43a-c0777af5fc44.json:1]: Fair point. We should probably revisit with the team what we want our policy to be. I don't think we really follow what we documented. I'll try to remember to bring it up at standup.

**Review by rajsite — APPROVED** [review]

**Pull request merged**